src/logging.rs

```diff
@@ -1,48 +1,76 @@
+use std::{fmt::Display, result::Result};
+
+#[derive(Clone)]
+struct FallbackToStderr<D: Drain> {
+    drain: D,
+}
+
+impl<D: Drain> Drain for FallbackToStderr<D>
+where
+    D::Err: Display,
+{
+    type Ok = ();
+    type Err = ();
+    fn log(&self, record: &Record, logger_values: &OwnedKVList) -> Result<(), ()> {
+        if let Err(err) = self.drain.log(record, logger_values) {
+            eprint!("A drain cloud not log to its destenation: {}", err);
+        }
+        Ok(())
+    }
+
+    #[inline]
+    fn is_enabled(&self, level: Level) -> bool {
+        self.drain.is_enabled(level)
+    }
+}
 
 pub fn create_logger(arg_matches: &ArgMatches) -> Result<slog::Logger, String> {
-    let min_log_level = match arg_matches.occurrences_of(args::VERBOSITY) {
-        0 => slog::Level::Warning,
-        1 => slog::Level::Info,
-        2 => slog::Level::Debug,
-        _ => slog::Level::Trace,
+    let min_log_level = match arg_matches.occurrences_of(VERBOSITY) {
+        0 => Level::Warning,
+        1 => Level::Info,
+        2 => Level::Debug,
+        _ => Level::Trace,
     };
 
-    let min_log_level = match arg_matches.value_of(args::LOG_LEVEL) {
+    let min_log_level = match arg_matches.value_of(LOG_LEVEL) {
         Some(level) => match level.parse::<args::LogLevelType>()? {
-            args::LogLevelType::error => slog::Level::Error,
-            args::LogLevelType::warn => slog::Level::Warning,
-            args::LogLevelType::info => slog::Level::Info,
-            args::LogLevelType::debug => slog::Level::Debug,
-            args::LogLevelType::trace => slog::Level::Trace,
+            args::LogLevelType::error => Level::Error,
+            args::LogLevelType::warn => Level::Warning,
+            args::LogLevelType::info => Level::Info,
+            args::LogLevelType::debug => Level::Debug,
+            args::LogLevelType::trace => Level::Trace,
         },
         None => min_log_level,
     };
 
-    let decorator = slog_term::TermDecorator::new().force_color().build();
-    let term_drain = slog_term::FullFormat::new(decorator)
+    let decorator = TermDecorator::new().force_color().build();
+    let term_drain = CompactFormat::new(decorator)
         .build()
         .filter_level(min_log_level)
+        .map(|drain| FallbackToStderr { drain })
         .fuse();
 
     let mut err: Option<String> = None;
     let drain = match redis_logger(&arg_matches) {
         Ok(Some(redis_logger)) => {
-            let both = slog::Duplicate::new(redis_logger, term_drain).fuse();
-            slog_async::Async::new(both.filter_level(min_log_level).fuse())
-                .build()
-                .fuse()
+            let both = Duplicate::new(redis_logger, term_drain).fuse();
+            Async::new(both.filter_level(min_log_level).fuse()).build().fuse()
         }
-        Ok(None) => slog_async::Async::new(term_drain).build().fuse(),
+        Ok(None) => Async::new(term_drain).build().fuse(),
```

Comment:
Even though it will probably have the same effect I think we should still use `ignore_res` over `fuse`.

Reply:
Just here or everywhere?

Reply:
everywhere since now we know it panicks right.

Reply:
I tried to make sure, that it will not happen, and I thought that it is better to panic and lose the thread and the logs, then to have no trace about that some logs got ignored. Did you consider this too, and you prefer to have logs silently ignored in case a programmer error (root drain not using `Err = ()`) in the log setup?

Reply:
Yeah OK, lets do it your way then I agree.